aws/provider_test.go

```diff
@@ -34,13 +34,20 @@ func init() {
 		"aws": testAccProvider,
 	}
 	testAccProviderFactories = func(providers *[]*schema.Provider) map[string]terraform.ResourceProviderFactory {
-		return map[string]terraform.ResourceProviderFactory{
-			"aws": func() (terraform.ResourceProvider, error) {
-				p := Provider()
-				*providers = append(*providers, p.(*schema.Provider))
+		// this is an SDKV2 compatible hack, the "factory" functions are
+		// effectively singletons for the lifecycle of a resource.Test
+		var providerNames = []string{"aws", "awseast", "awswest", "awsalternate", "awsus-east-1", "awsalternateaccountalternateregion", "awsalternateaccountsameregion", "awssameaccountalternateregion", "awsthird"}
+		var factories = make(map[string]terraform.ResourceProviderFactory, len(providerNames))
+		var singletons = make(map[string]*schema.Provider, len(providerNames))
```

Comment:
this doesn't seem to get used anywhere

Reply:
Oh right... i guess somewhere in the refactoring i lost the need for it

Reply:
+1 removing this 👍